esp/logging/loggingmanager/loggingmanager.cpp

```diff
@@ -250,6 +262,11 @@ bool CLoggingManager::updateLog(IEspContext* espContext, IEspUpdateLogRequestWra
                 IUpdateLogThread* loggingThread = loggingAgentThreads[x];
                 if (loggingThread->hasService(LGSTUpdateLOG))
                 {
+                    //leave the fact that a script can control the thread queue as an option undocumented, naming scheme could change,
+                    //  controlling the queue is a very low level mechanism and should be frowned upon
+                    //  once scripts can communicate with the agent that should be the mechanism to skip a particular type of logging
```

Comment:
Maybe it's just semantics, but this comment rubs me the wrong way. As long as the manager is driving the individual agents (i.e., not decoupled), it seems like controlling the queue is a mechanism that should be embraced as an optimization rather than be frowned upon. I would replace this mechanism, which is very low level, with one that looks at the agent behind the thread and compares the agent's types and groups with selected profiles and options to decide if any agent type is needed. In other words, replace explicit queue control with implicit control based on the same information the agent will use to control its own behavior.

Reply:
I thought you already had a JIRA for adding that capability.  Once we have that we shouldn't need to use this queue centric approach.  And once we support decoupled logging the queue wont directly correspond to an agent anyway.  The sooner your changes are available the better.  Do we need to create an additional JIRA?

Reply:
There is a JIRA to make the agent pay attention to the information. The point I've unsuccessfully tried to make is that a faster path to a solution exists. The log thread owns a log agent. The log agents that we need to control now can specify types and groups, and all agents support the interface to access this data. The long term solution is to use profile (groups) and options (types) in the agent. A short term solution can use profile and options in the exact same place you're changing to accomplish the same outcome.

The queue control is a sufficient solution until we have requirements to either disable part of an agent or use an agent in multiple profiles. The one use case I know of for one agent supporting multiple types does not apply, and I don't know of any requirement to use an agent with multiple profiles.

Reply:
How does the log manager ask for the agents types and groups?  If it's already exposed I can look into checking it.. I just don't want to make any log agent changes or even extensive log manager changes in this PR.  If the info is already exposed let me know how to access it and I'll see what I can do.

Reply:
The agent group (not agent type) is introduced in Decoupled Logging service. It has not been implemented for the logging manager in D-ESDL side. We may implement it after this PR.

Reply:
virtual IEspLogAgentVariantIterator* getVariants() const = 0;
This agent interface method return an iterator of variants, which are triplets of agent name, type, and group. All agents should be responding with at least one variant (name, "", ""). The scapps migration agent can be configured with a list of type and group. Each list item maps to a variant in the iterator.

Other agents should be able to add support by updating their configuration schemas. The code to process the configuration is in the agent base class.

Reply:
I am making the change to check group in the log manager.  I expect you will do whatever it takes to package up the scriptValues so that the various logging agents have access to them locally and remotely later.  But my change should let us control by supported group names (aka profile name).